src/whitenoise/accounts.rs

```diff
@@ -857,10 +849,77 @@ impl Whitenoise {
             );
 
             let current_time = Utc::now().timestamp_millis();
+
+            let whitenoise = match Whitenoise::get_instance() {
+                Ok(instance) => instance,
+                Err(e) => {
+                    tracing::error!(
+                        target: "whitenoise::background_fetch_account_data",
+                        "Failed to get Whitenoise instance for background sync: {}",
+                        e
+                    );
+                    return;
+                }
+            };
+
+            // Get user's relay information for contact list fetching
+            let user_relays = match account_clone.nip65_relays(whitenoise).await {
+                Ok(relays) => relays.into_iter().map(|r| r.url).collect::<Vec<_>>(),
+                Err(_) => {
+                    // Fallback to default relays if user has no specific relays
+                    Relay::defaults()
+                        .into_iter()
+                        .map(|r| r.url)
+                        .collect::<Vec<_>>()
+                }
+            };
+
+            // Ensure user relays are connected before fetching contact list
+            if let Err(e) = whitenoise.nostr.ensure_relays_connected(&user_relays).await {
+                tracing::warn!(
+                    target: "whitenoise::background_fetch_account_data",
+                    "Failed to ensure relay connections for account {}: {}. Proceeding with fetch anyway.",
+                    account_clone.pubkey.to_hex(),
+                    e
+                );
+            }
+
+            // Fetch the account's contact list
+            let fetched_contact_list = whitenoise
+                .nostr
+                .fetch_contact_list_events(account_clone.pubkey, &user_relays)
+                .await;
```

Comment:
I think this logic should be within the `fetch_user_event_streams` method (or within a similar, dedicated one) in NostrManager. It should receive the vector of relay urls.

This would:
a) Make sure we have a signer set
b) Run in parallel with other relay event fetching